lib/nerve/reporter/base.rb

```diff
@@ -19,10 +19,18 @@ def report_up
     def report_down
     end
 
-    def update_data(new_data='')
+    def ping?
     end
 
-    def ping?
+    def get_service_data(service)
+      %w{instance_id host port}.each do |required|
+        raise ArgumentError, "missing required argument #{required} for new service watcher" unless service[required]
+      end
+      d = {'host' => service['host'], 'port' => service['port'], 'name' => service['instance_id']}
+      if service['weight'].to_i >= 0 and "#{service['weight']}".match /^\d+$/
```

Comment:
oh i see this is only run once, on startup. in either case, i think it's preferred to fail than to get an unexpected result